File: ios/WebViewWithRefreshManager.m
//  Created by react-native-create-bridge

#import <Foundation/Foundation.h>
#import "WebViewWithRefresh.h"
#import "WebViewWithRefreshManager.h"

// import RCTBridge
#if __has_include(<React/RCTBridge.h>)
#import <React/RCTBridge.h>
#elif __has_include(“RCTBridge.h”)
#import “RCTBridge.h”
#else
#import “React/RCTBridge.h” // Required when used as a Pod in a Swift project
#endif


@implementation WebViewWithRefreshManager {
  WebViewWithRefresh *_wv;
}

@synthesize bridge = _bridge;

// Export a native module
// https://facebook.github.io/react-native/docs/native-modules-ios.html
RCT_EXPORT_MODULE();

// Return the native view that represents your React component
- (UIView *)view
{
  _wv = [[WebViewWithRefresh alloc] initWithEventDispatcher:self.bridge.eventDispatcher];
  return _wv;
}
//
RCT_EXPORT_VIEW_PROPERTY(url, NSString)
RCT_EXPORT_VIEW_PROPERTY(matchCondition, NSString)
RCT_EXPORT_VIEW_PROPERTY(onUrlMatch, RCTBubblingEventBlock)
RCT_EXPORT_METHOD(goTo:(NSString*)url) {
  [_wv goto:url];
}

@end

File: android/src/main/java/com/reactlibrary/WebViewWithRefreshManager.java
//  Created by react-native-create-bridge

package com.reactlibrary;

import android.support.annotation.Nullable;
import android.view.View;

import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.common.MapBuilder;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;

import com.facebook.react.uimanager.annotations.ReactProp;
import android.webkit.WebView;
import android.webkit.WebViewClient;

import java.util.Map;

public class WebViewWithRefreshManager extends SimpleViewManager<View> {
    public static final String REACT_CLASS = "WebViewWithRefresh";

    @Override
    public String getName() {
        // Tell React the name of the module
        // https://facebook.github.io/react-native/docs/native-components-android.html#1-create-the-viewmanager-subclass
        return REACT_CLASS;
    }

    @Override
    public WebViewWithRefresh createViewInstance(ThemedReactContext context){
        // Create a view here
        // https://facebook.github.io/react-native/docs/native-components-android.html#2-implement-method-createviewinstance
        WebViewWithRefresh wv = new WebViewWithRefresh(context);
        return wv;
    }

    @ReactProp(name = "url")
    public void setUrl(WebViewWithRefresh view, String prop) {
        // Set properties from React onto your native component via a setter method
        // https://facebook.github.io/react-native/docs/native-components-android.html#3-expose-view-property-setters-using-reactprop-or-reactpropgroup-annotation
        view.goTo(prop);
    }

    @ReactProp(name = "matchCondition")
    public void setMatchCondition(WebViewWithRefresh view, String condition) {
        view.setCondition(condition);
    }

    @Override
    public @Nullable
    Map getExportedCustomDirectEventTypeConstants() {
        return MapBuilder.<String, Object>builder()
            .put("onUrlMatch",
                    MapBuilder.of("registrationName", "onUrlMatch"))
            .put("onStartLoad",
                    MapBuilder.of("registrationName", "onStartLoad"))
            .put("onFinishLoad",
                    MapBuilder.of("registrationName", "onFinishLoad"))
            .build();
    }



}
